critwall/src/main/java/org/opencadc/critwall/FileSyncJob.java

```diff
@@ -126,16 +131,32 @@ public FileSyncJob(URI artifactID, URI locatorServiceID, StorageAdapter storageA
         this.artifactDAO = artifactDAO;
     }
 
+    public void setOwner(final Subject owner) {
+        this.owner = owner;
+    }
+
     @Override
     public void run() {
         log.info("fileSyncJob start.");
+        if (this.owner == null) {
+            doSync();
+        } else {
+            Subject.doAs(this.owner, (PrivilegedAction<Void>) () -> {
+                doSync();
+                return null;
```

Comment:
Should not have a `return null` here. Surprised compiler didn't complain...

Reply:
The `PrivilegedAction` needs to return something though.  Is there a different way to do it?

Reply:
Actually, changed to be a `RunnableAction` instead.